contributor-docs/adrs/adr-xxx-behavior-isolation.md

```diff
@@ -0,0 +1,137 @@
+
+#### SSR
+
+Server Side Rendering, or "SSR" refers to construcing templates of HTML strings on the server side and delivering HTML. Custom Elements can be server rendered by specifying their tag name in the markup delivered, but they have no constraints around rendering on the server, as their behavior is entirely client side driven. Frameworks for the server such as Ruby on Rails can use "partials" or View Components to represent a custom elements markup for the server.
+
+React's core library does not offer a way to represent React's component tree in either the real DOM, nor on the server. In this way, developers can use either the `react-dom` library on the client, or the `react-dom/server` library using a runtime like NodeJS on the server to deliver a similar experience to traditional server side applications built in frameworks like Ruby on Rails. This is sometimes referred to as "isomorphic" code. The unique selling point of "isomorphic" components, is that developers can write a component once and have it execute in both environments, delivering a string of HTML in a server response, and reconciling DOM operations on the client.
+
+ADR 002 describes complexity around Custom Elements and Server Side Rendering. It claims that for Custom Elements to be well supported, that Primer React would need to accumulate added complexity which would be surfaced to the user, going on to claim that while libraries may exist their usage is antithetical to Custom Element patterns.
+
+Primer React's [recent addition of the Relative Time component](https://github.com/primer/react/pull/2484) (itself a wrapper of the [`relative-time`](https://github.com/github/relative-time-element) custom element) demonstrates that a custom element can be loaded into a server side runtime such as NodeJS without causing error, by adding [2 additional lines of code to the element itself](https://github.com/github/relative-time-element/blob/4ab3f2807a02dfd687b2b7fa04553300ff87ddbe/src/relative-time-element.ts#L2-L3):
+
+```typescript
+const root = (typeof globalThis !== 'undefined' ? globalThis : window) as typeof window
+const HTMLElement = root.HTMLElement || (null as unknown as typeof window['HTMLElement'])
+```
```

Comment:
While this code is indeed compact, there a few caveats that should be considered

- This code needs to be included in _every_ custom element definition. As you state below, this could potentially be provided globally in the SSR environment, but we still need to code present in case it isn't provided globally.
- This isn't the only code needed for SSR safety - there is also [this chunk](https://github.com/github/relative-time-element/blob/main/src/relative-time-element-define.ts#L3-L14) which is responsible for registering the custom element and safely falling back to doing _nothing_ if the environment doesn't support custom elements. Compare with [this other example](https://github.com/github/clipboard-copy-element/blob/main/src/index.ts#L14-L17), which is our standard registration code.